include/shadesmar/robust_lock.h

```diff
@@ -66,6 +68,26 @@ void RobustLock::lock() {
   exclusive_owner = getpid();
 }
 
+bool RobustLock::try_lock() {
+  if (exclusive_owner != 0) {
```

Comment:
You've split it into two parts, but that's not needed. 

Take the case where there's no lock. (`exclusive_owner = 0`), so you'll call `prune_sharable_procs()` which is not needed, so doing a `try_lock()` at the start will make it much faster. 

It should be something like this:
```c++
if (!mutex_.try_lock()) {
  // mutex_ is held. Do the check for dead procs here. 
} else {
  exclusive_owner = getpid();
  return true;
}
```

Reply:
But after we `try_lock()` and it doesn't work and we check for dead procs and remove those locks we should `try_lock()` again right? I agree `try_lock()`at the beginning will be much faster but do we `try_lock()` again after pruning or checking for dead procs?

Reply:
Yup, you'll do a check again. We'll be introducing an extra branch, but "Robust" mutexes won't be used for opt builds anyway, it's only for safety guaranteed.